addons/web_editor/controllers/main.py

```diff
@@ -287,20 +287,30 @@ def _attachment_create(self, name='', data=False, url=False, res_id=False, res_m
         else:
             res_id = False
 
-        attachment_data = {
-            'name': name,
-            'public': res_model == 'ir.ui.view',
-            'res_id': res_id,
-            'res_model': res_model,
-        }
```

Comment:
Why did you mov `attachment_data` inside the `if/elif` ? Because now well it is duplicated, which is not good.

Reply:
@dmo-odoo i looked into all the changes you have questioned, i will update the code according to it. and remove any unnecessary changes. Thanks !